backend/shared/shared/oidc/views/eauth_views.py

```diff
@@ -92,10 +97,22 @@ class EauthAuthenticationCallbackView(View):
     http_method_names = ["get"]
 
     def login_success(self):
-        return HttpResponseRedirect(settings.LOGIN_REDIRECT_URL)
+        url = settings.LOGIN_REDIRECT_URL
+
+        lang = self.request.session.pop("lang", None)
+        if lang:
+            url = f"{url}/{lang}/"
+
+        return HttpResponseRedirect(url)
 
     def login_failure(self):
-        return HttpResponseRedirect(settings.LOGIN_REDIRECT_URL_FAILURE)
+        url, error_path = settings.LOGIN_REDIRECT_URL_FAILURE.rsplit("/", 1)
+
+        lang = self.request.session.pop("lang", None)
+        if lang:
+            url = f"{url}/{lang}/{error_path}"
```

Comment:
I assume this URL format has been agreed with UI devs?

Reply:
Yes